src/downloader.rs

```diff
@@ -225,21 +243,174 @@ impl Default for DownloaderBuilder {
+    /// Create new [`Downloader`] [`StyleOptions`].
+    pub fn new(main: ProgressBarOpts, child: ProgressBarOpts) -> Self {
+        Self { main, child }
+    }
+
+    /// Set the options for the main progress bar.
+    pub fn set_main(&mut self, main: ProgressBarOpts) {
+        self.main = main;
+    }
+
+    /// Set the options for the child progress bar.
+    pub fn set_child(&mut self, child: ProgressBarOpts) {
+        self.child = child;
+    }
+}
+
+/// Define the options for a progress bar.
+#[derive(Debug, Clone)]
+pub struct ProgressBarOpts {
+    /// Progress bar template string.
+    template: Option<String>,
+    /// Progression characters set.
+    ///
+    /// There must be at least 3 characters for the following states:
+    /// "filled", "current", and "to do".
+    progress_chars: Option<String>,
+    /// Enable or disable the progress bar.
+    enabled: bool,
+    /// Clear the progress bar once completed.
+    clear: bool,
+}
+
+impl Default for ProgressBarOpts {
+    fn default() -> Self {
+        Self {
+            template: None,
+            progress_chars: None,
+            enabled: true,
+            clear: true,
+        }
+    }
+}
+
+impl ProgressBarOpts {
+    /// Template representing the bar and its position.
+    ///
+    ///`███████████████████████████████████████ 11/12 (99%) eta 00:00:02`
+    pub const TEMPLATE_BAR_WITH_POSITION: &'static str =
+        "{bar:40.blue} {pos:>}/{len} ({percent}%) eta {eta_precise:.blue}";
+    /// Template which looks like the Python package installer pip.
+    ///
+    /// `━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 211.23 KiB/211.23 KiB 1008.31 KiB/s eta 0s`
+    pub const TEMPLATE_PIP: &'static str =
+        "{bar:40.green/black} {bytes:>11.green}/{total_bytes:<11.green} {bytes_per_sec:>13.red} eta {eta:.blue}";
+    /// Use increasing quarter blocks as progress characters: `"█▛▌▖  "`.
+    pub const CHARS_BLOCKY: &'static str = "█▛▌▖  ";
+    /// Use fade-in blocks as progress characters: `"█▓▒░  "`.
+    pub const CHARS_FADE_IN: &'static str = "█▓▒░  ";
+    /// Use fine blocks as progress characters: `"█▉▊▋▌▍▎▏  "`.
+    pub const CHARS_FINE: &'static str = "█▉▊▋▌▍▎▏  ";
+    /// Use a line as progress characters: `"━╾─"`.
+    pub const CHARS_LINE: &'static str = "━╾╴─";
+    /// Use rough blocks as progress characters: `"█  "`.
+    pub const CHARS_ROUGH: &'static str = "█  ";
+    /// Use increasing height blocks as progress characters: `"█▇▆▅▄▃▂▁  "`.
+    pub const CHARS_VERTICAL: &'static str = "█▇▆▅▄▃▂▁  ";
+
+    /// Create a new [`ProgressBarOpts`].
+    pub fn new(
+        template: Option<String>,
+        progress_chars: Option<String>,
+        enabled: bool,
+        clear: bool,
+    ) -> Self {
+        Self {
+            template,
+            progress_chars,
+            enabled,
+            clear,
+        }
+    }
+
+    /// Create a [`ProgressStyle`] based on the provided options.
+    pub fn to_progress_style(self) -> ProgressStyle {
```

Comment:
You could probably do this as `impl From<ProgressBarOpts> for ProgressStyle`

Reply:
```diff
- let style = self.to_progress_style(); 
+ let style = self.into(); 
 ```
https://github.com/rgreinho/trauma/blob/8f5b3986785f0963f6018c05bcf48050449147c2/src/downloader.rs#L383

Reply:
I thought of it, but then, I thought that `impl From<T>` implies a conversion. However here we are using some values from the options to build a ProgressStyle. Therefore It did not seem very idiomatic to me.

But maybe I was thinking about it too much. If you still don't see any problem after reading this, I'll implement the 
`From` trait.